src/nvim/os/fs.c

```diff
@@ -169,6 +181,37 @@ static bool is_executable_in_path(const char_u *name, char_u **abspath)
       return true;
     }
 
+#ifdef WIN32
+    // Try appending file extensions from $PATHEXT to the name
+    char *buf_end = xstrchrnul((char *)buf, '\0');
+    for (const char *ext = pathext; *ext; ext++) {
+      if (ext[0] != '.' || ext[1] == '\0') {
+        // Skip if there is no suffix after '.' or
+        // if it does not start with '.'
```

Comment:
Why are we skipping the whole thing if it doesn't start with a dot? [The Vim logic](https://github.com/neovim/neovim/pull/3291#issuecomment-137979590) makes a special case for a single dot, but otherwise just tries up to the next `;`. 

```c
    p = mch_getenv("PATHEXT");
    if (p == NULL)
        p = (char_u *)".com;.exe;.bat;.cmd";
    while (*p)
    {
        if (p[0] == '.' && (p[1] == NUL || p[1] == ';'))
        {
            /* A single "." means no extension is added. */
            buf[len] = NUL;
            ++p;
            if (*p)
                ++p;
        }
        else
            copy_option_part(&p, buf + len, _MAX_PATH - len, ";");
        if (executable_exists((char *)buf, path))
            return TRUE;
    }
```